internal/pkg/cli/env_init.go

```diff
@@ -297,32 +280,55 @@ func (o *initEnvOpts) askEnvName() error {
 	return nil
 }
 
-func (o *initEnvOpts) askEnvProfile() error {
-	// TODO: add this behavior to selector pkg.
+func (o *initEnvOpts) askEnvSession() error {
 	if o.Profile != "" {
+		sess, err := o.sessProvider.FromProfile(o.Profile)
+		if err != nil {
+			return fmt.Errorf("create session from profile %s: %w", o.Profile, err)
+		}
+		o.sess = sess
 		return nil
 	}
-
-	names := o.profileConfig.Names()
-	if len(names) == 0 {
-		return errNamedProfilesNotFound
+	if o.TempCreds.isSet() {
+		sess, err := o.sessProvider.FromStaticCreds(o.TempCreds.AccessKeyID, o.TempCreds.SecretAccessKey, o.TempCreds.SessionToken)
+		if err != nil {
+			return err
+		}
+		o.sess = sess
+		return nil
 	}
-
-	profile, err := o.prompt.SelectOne(
-		fmt.Sprintf(fmtEnvInitProfilePrompt, color.HighlightUserInput(o.Name)),
-		envInitProfileHelpPrompt,
-		names)
+	sess, err := o.selCreds.Creds(fmt.Sprintf(fmtEnvInitCredsPrompt, color.HighlightUserInput(o.Name)), envInitCredsHelpPrompt)
 	if err != nil {
-		return fmt.Errorf("get the profile name: %w", err)
+		return fmt.Errorf("select creds: %w", err)
+	}
+	o.sess = sess
+	return nil
+}
+
+func (o *initEnvOpts) askEnvRegion() error {
+	region := aws.StringValue(o.sess.Config.Region)
+	if o.Region != "" {
+		region = o.Region
+	}
+	if region == "" {
+		v, err := o.prompt.Get(envInitRegionPrompt, "", nil, prompt.WithDefaultInput(envInitDefaultRegionOption))
+		if err != nil {
+			return fmt.Errorf("get environment region: %w", err)
+		}
+		region = v
 	}
-	o.Profile = profile
+	o.sess.Config.Region = aws.String(region)
 	return nil
 }
 
 func (o *initEnvOpts) askCustomizedResources() error {
 	if o.NoCustomResources {
 		return nil
 	}
+	if o.selVPC == nil {
```

Comment:
Is there any chance that this `o.selVPC` is not nil?

Reply:
The nil checks are for testing or the `init` command which sets these dependencies on its own